R/add.R

```diff
@@ -142,6 +142,66 @@ m_add_curve <- function(id, spec = list()) {
 #' @param toCap Cap at end of cylinder. 0 for none, 1 for flat, 2 for rounded.
 #' @param dashed Boolean, dashed style cylinder instead of solid.
 #' @param spec Additional shape specifications defined with
+#' @param color Color value for cylinder.
+#' \code{m_shape_spec()}.
+#' @noRd
+#' @keywords internal
+.m_add_cylinder <- function(
+                            id,
+                            start,
+                            end,
+                            radius = 0.1,
+                            fromCap = 1,
+                            toCap = 1,
+                            dashed = FALSE,
+                            color,
+                            hidden,
+                            wireframe,
+                            alpha,
+                            spec = m_shape_spec()) {
+  arglist <- list(
+    start = start,
+    end = end,
+    radius = radius,
+    fromCap = fromCap,
+    toCap = toCap,
+    dashed = dashed,
+    color = color,
+    hidden = hidden,
+    wireframe = wireframe,
+    alpha = alpha
+  )
+
+  spec <- c(arglist, spec)
+
+  method <- "addCylinder"
+  callJS()
+}
+
+#' Add Cylinder Between Points
+#'
+#' Add cynliders between the given points. Will match starting point/s with
+#' ending point/s to create a line between each point. Styling options can be
+#' supplied as one option, or a vector of length equal to the number of lines.
+#' @param id R3dmol \code{id} or a \code{r3dmol} object (the output from
+#' \code{r3dmol()}).
+#' @param start Starting position (or \code{list()} of positions) of line. Can
+#' be a single position or \code{list()} of positions. Format either
+#' \code{m_sel()} or \code{m_vector3()}.
+#' @param end Ending position (or \code{list()} of positions) of line. Can
+#' be a single position or \code{list()} of positions. Format either
```

Comment:
The usage of multiple atoms is still a little difficult for me to grasp intuitively. It would be better If you could add a more example to show that how to use the multiple start(s) and end(s).  Also try to avoid using `m_fetch_pdb` in the example, it will use the network when running the check, so it will take longer to check and may cause errors due to the network.

And an article about how to use selection / multiple selection, add multiple shape at once would be prefer (Could be done in another PR).

Reply:
I had added some additional examples for `m_add_cylinder()` and will write up a going through the multiple selections / multi-shape add.